package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -800,8 +885,14 @@ def count_by_type(self):
         d = self.u.atoms[self.results.hbonds[:, 1].astype(np.intp)]
         a = self.u.atoms[self.results.hbonds[:, 3].astype(np.intp)]
 
-        tmp_hbonds = np.array([d.resnames, d.types, a.resnames, a.types],
-                              dtype=str).T
+        if hasattr(d, "resnames"):
+            d_res = d.resnames
+            a_res = a.resnames
+        else:
+            d_res = [None for x in range(len(d.types))]
```

Comment:
simplify to
```python
d_res = len(d.types) * [None]
a_res = len(a.types) * [None]
```

Reply:
Given that this is turned into `str`, replace `None` with `"None"`?? It's twice as fast for me
```python
In [5]: a = 1000*[None]

In [6]: %timeit np.array(a, dtype=str)
255 µs ± 2.47 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

In [7]: b = 1000*["None"]

In [8]: %timeit np.array(b, dtype=str)
128 µs ± 561 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```
although it's already fast so perhaps clarity of using `None` is better than introducing magic-looking strings.

However, given that these resnames are not really necessary for anything, why not just use `np.zeros(len(a.types), dtype=str)`
```python
In [10]: %timeit np.zeros(1000, dtype=str)
412 ns ± 4.31 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

Reply:
I changed it to use "None", thanks for the benchmarking. I think the suggest behavior of using zero could lead a user to believe that their whole system is important in with a uniform integer as a rename. While, use of "None" is unambiguous.